ddtrace/contrib/trace_utils.py

```diff
@@ -69,49 +69,83 @@ def int_service(pin, config, default=None):
-        return config.service_name
+    if "service" in int_config and int_config.service is not None:
+        return int_config.service
+    if "service_name" in int_config and int_config.service_name is not None:
+        return int_config.service_name
 
-    global_service = config.global_config._get_service()
+    global_service = int_config.global_config._get_service()
     if global_service:
         return global_service
 
-    if "_default_service" in config and config._default_service is not None:
-        return config._default_service
+    if "_default_service" in int_config and int_config._default_service is not None:
+        return int_config._default_service
 
     return default
 
 
-def ext_service(pin, config, default=None):
+def ext_service(pin, int_config, default=None):
     """Returns the service name for an integration which is external
     to the application. External meaning that the integration generates
     spans wrapping code that is outside the scope of the user's application. Eg. A database, RPC, cache, etc.
     """
-    config = config or {}
+    int_config = int_config or {}
 
     if pin and pin.service:
         return pin.service
 
-    if "service" in config and config.service is not None:
-        return config.service
-    if "service_name" in config and config.service_name is not None:
-        return config.service_name
+    if "service" in int_config and int_config.service is not None:
+        return int_config.service
+    if "service_name" in int_config and int_config.service_name is not None:
+        return int_config.service_name
 
-    if "_default_service" in config and config._default_service is not None:
-        return config._default_service
+    if "_default_service" in int_config and int_config._default_service is not None:
+        return int_config._default_service
 
     # A default is required since it's an external service.
     return default
 
 
-def set_http_meta(span, config, method=None, url=None, status_code=None):
+def is_error_code(status_code):
+    """Returns a boolean representing whether or not a status code is an error code.
+    Error status codes by default are 500-599.
+    You may also enable custom error codes::
+
+        from ddtrace import config
+        config.http_server.error_statuses = '401-404,419'
+
+    Ranges and singular error codes are permitted and can be separated using commas.
+    """
+    try:
+        error_str = config.http_server.error_statuses
+        error_ranges = error_str.split(",")
+        for error_range in error_ranges:
+            values = error_range.split("-")
+            min_code = int(values[0])
+
+            if len(values) == 2:
+                max_code = int(values[1])
+            else:
+                max_code = min_code
+            if min_code > max_code:
+                tmp = min_code
+                min_code = max_code
+                max_code = tmp
+
+            if min_code <= int(status_code) <= max_code:
```

Comment:
Could be simplified perhaps to:

```suggestion
            values = [int(v) for v in values]
            if min(values) <= int(status_code) <= max(values):
```